schema-nft-marketplace.graphql

```diff
@@ -0,0 +1,255 @@
+    network: Network!
+
+    " Version of the subgraph schema, in SemVer format (e.g. 1.0.0) "
+    schemaVersion: String!
+
+    " Version of the subgraph implementation, in SemVer format (e.g. 1.0.0) "
+    subgraphVersion: String!
+
+    " Version of the methodology used to compute metrics, loosely based on SemVer format (e.g. 1.0.0) "
+    methodologyVersion: String!
+
+    " Number of collections listed on the marketplace. "
+    collectionCount: Int!
+
+    " Trade count of the all collections on the marketplace. "
+    tradeCount: Int!
+
+    " Cumulative trade volume (in ETH) "
+    cumulativeTradeVolumeETH: BigDecimal!
+
+    " Revenue that goes to the marketplace protocol, aka protocol fee. "
+    marketplaceRevenueETH: BigDecimal!
+
+    " Revenue that goes to creator, aka royalty fee. "
+    creatorRevenueETH: BigDecimal!
+
+    " Sum of marketplaceRevenueETH and creatorRevenueETH. "
+    totalRevenueETH: BigDecimal!
+
+    " Cumulative unique traders "
+    cumulativeUniqueTraders: Int!
+}
+
+type Collection @entity {
+    " Contract address. "
+    id: ID!
+
+    " Collectio name, mirrored from the smart contract. Leave null if not available. "
+    name: String
+
+    " Collection symbol, mirrored from the smart contract. Leave null if not available. "
+    symbol: String
+
+    " Total supply of the collection, mirrored from the smart contract. "
+    totalSupply: BigInt!
+
+    " NFT Standard the collection uses. "
+    nftStandard: NftStandard!
+
+    " Royalty fee rate in percentage. E.g. 2.5% should be 2.5 "
+    royaltyFee: BigDecimal!
+
+    " Number of tokens listed for trade in the marketplace. "
+    listedCount: BigInt!
+
+    " Floor price (in ETH) "
+    floorPriceETH: BigDecimal!
+
+    " Cumulative trade volume (in ETH) "
+    cumulativeTradeVolumeETH: BigDecimal!
+
+    " Revenue that goes to the marketplace protocol, aka protocol fee. "
+    marketplaceRevenueETH: BigDecimal!
+
+    " Revenue that goes to creator, aka royalty fee. "
+    creatorRevenueETH: BigDecimal!
+
+    " Sum of marketplaceRevenue and creatorRevenue. "
+    totalRevenueETH: BigDecimal!
+
+    " Trade count of the collection on the marketplace. "
+    tradeCount: Int!
+
+    " Trades of the collection. "
+    trades: [Trade!]! @derivedFrom(field: "collection")
+}
+
+" Trades exist such as a combination of taker/order and bid/ask. "
+type Trade @entity {
+    " { maker txn hash }-{ taker txn hash } "
+    id: ID!
+
+    " Transaction hash of the maker. "
+    makerTransactionHash: String!
+
+    " Transaction hash of the taker. "
+    takerTransactionHash: String!
+
+    " Event log index. "
+    logIndex: Int!
+
+    " Block timestamp where the trade is executed. "
+    timestamp: BigInt!
+
+    " Block number where the trade is executed. "
+    blockNumber: BigInt!
+
+    " Collection involved "
+    collection: Collection!
+
+    " Token ID of the traded NFT. "
+    tokenId: BigInt!
```

Comment:
why don't we create a separate entity for NFT, which can contain all the metadata-related information, and information about the owner, creator, number of transfers, and holding period(for how long the user is holding the NFT) of the NFT.

Reply:
yes, we will have a separate subgraph for NFT holder data and NFT metadata